### DISCUSSION TITLE
AttachAction in v3, $titleAttribute vs. recordTitle()

### DISCUSSION THREAD

**Discussion by ajnsn** [filamentphp/filament#7568] [Help] [answered]
In v2 it was possible to have a $recordTitleAttribute on the RelationManager together with a custom recordTitle() on the AttachAction.

In v3 there's this if-clause (currently on line [268](https://github.com/filamentphp/filament/blob/3.x/packages/tables/src/Actions/AttachAction.php#L268)) which prefers the  $titleAttribute to the mapping with getRecordTitle() (currently on line [278](https://github.com/filamentphp/filament/blob/3.x/packages/tables/src/Actions/AttachAction.php#L278)) 

Took me a while to find out and it seems to be unwanted, at least from my side. The RelationManager's should be the fallback, not the priority.  This specifically prevents the ability to show HTML labels with the record select. Or is there any other way to achieve this in v3?

Thanks

**Comment by danharrin** [comment]
Sure, it's unwanted, but that if statement does need to be there to ensure that we are plucking where possible. We need to add another if statement before 268 that checks to see if a custom record title is set for that action or the table, I guess?

**Reply by ajnsn** [reply]
Thank you for you fast answer @danharrin. Does the `getRecordTitle()` already fallback to `getTableRecordTitle()` of the RelationManager? If so, we do not need to 268-if-clause as v2 does it?

**Reply by danharrin** [reply]
Correct, `getRecordTitle()` falls back to `getTableRecordTitle()`, but if we use `getRecordTitle()` then we *must* use `get()` on the query to fully hydrate models, instead of `pluck()`. Which is a performance issue. So unless there is a closure to resolve the record title, we should be checking the attribute and plucking. Only `get()` if we really have to.

**Reply by ajnsn** [reply]
Ah, agree. But there more occurences of `getRecordTitle()`, e.g. Line [308](https://github.com/filamentphp/filament/blob/3.x/packages/tables/src/Actions/AttachAction.php#L308) when generating the optionLabel, so I see inconsistencies here. :)

I am also unsure how to PR this, so maybe you have an idea how to solve this.

**Reply by danharrin** [reply]
I'm not going to be able to prioritise making a PR on this any time soon, since we have quite a lot of critical v3 bugs and this is more of an inconvenience. One way to fast-track a bug fix is to sponsor it, for example booking a [consultation](https://filamentphp.com/consulting) where we can look at the issue together and PR a fix.

**Reply by ajnsn** [reply]
No worries, yes it breaks the upgrade to v3 here as we do have a lot of "person records" which do have identically names and having additional information (like city) is mandatory for users. I do also sponsor you since  February 26, 2022

**Reply by danharrin** [reply]
Yeah and I do really appreciate the sponsorship, but the time it would take to work out a nice solution would be time that I probably should be spending on a more critical bug. I am working burnt-out 13 hour days right now and literally can't do any more. Feel free to open an issue with a reproduction repo so it stays on my radar, but I can't guarantee a date that it will be fixed by.

**Comment by danharrin** [comment, marked as answer]
Should be fixed by #8992
